File: ecommerce-go-fiber/apps/users/internal/config/env.go
package config

import (
	"os"
)

type Env struct {
	UserPort        string
	UserMongoURI    string
	UserMongoDBName string
	UserCollName    string
}

func getEnv(key, fallback string) string {
	value := os.Getenv(key)
	if value == "" {
		return fallback
	}
	return value
}

func LoadEnv() Env {

	return Env{
		UserPort:        getEnv("PORT", "4010"),
		UserMongoURI:    getEnv("MONGO_URI", "mongodb://admin:password@localhost:27017"),
		UserMongoDBName: getEnv("MONGO_DB_NAME", "users_db"),
		UserCollName:    getEnv("COLLECTION_NAME", "users"),
	}
}

File: ecommerce-ts-express/services/users/src/config/env.ts
export type Env = {
  port: number
  mongoUri: string
  serviceName: string
}

export function getEnv(): Env {
  return {
    port: Number(process.env.PORT ?? 3010),
    mongoUri:
      process.env.MONGO_URI ??
      'mongodb://admin:password@localhost:27017/users_db?authSource=admin',
    serviceName: process.env.SERVICE_NAME ?? 'users'
  }
}
